dev/breeze/src/airflow_breeze/commands/developer_commands.py

```diff
@@ -357,17 +351,12 @@ def start_airflow(
 def build_docs(
     docs_only: bool,
     spellcheck_only: bool,
-    for_production: bool,
     builder: str,
     clean_build: bool,
     one_pass_only: bool,
     package_filter: tuple[str],
     github_repository: str,
 ):
-    """Build documentation in the container."""
-    if for_production and not clean_build:
-        get_console().print("\n[warning]When building docs for production, clean-build is forced\n")
-        clean_build = True
```

Comment:
Just double checking, `for_production` is now by default, before we were forcing `clean-build` when `for_production` was `True`. Does it make sense to keep that option then? Now we can have `clean_build = False` although `for_production` is like `True`

Reply:
Yes. That's good point and keen eye :) . I've thought about it to but we are pretty well covered here.

The  `--clean-build` flag makes rebuilding docs slow (because it deletes all the cache) - so we certainly do not want to have it enabled by default. And the "force clean build for production" was added there mostly as a protection against somoene mistakenly building docs without `--clean-build` while releasing it and possibly publishing some old documentation.

However since it was introduced, we've also been very keen on perfecting our release process and we largely simply copy&paste the steps from here:

* https://github.com/apache/airflow/blob/main/dev/README_RELEASE_AIRFLOW.md#publish-documentation
* https://github.com/apache/airflow/blob/main/dev/README_RELEASE_HELM_CHART.md#publish-documentation
* https://github.com/apache/airflow/blob/main/dev/README_RELEASE_PROVIDER_PACKAGES.md#prepare-documentation

As you can check (and I double checked it before submitting this PR) in all cases there we have `--clean-build` added as a flag. Yes. it is not perfect, someone impatient might want to skip the copy&paste step but we really focus in our release management team on following the release process "by-the-book" and I think we are all aware that we should not do shortcuts. Also the risk is small, even if build is not clean, the probability of breaking docs is very small - and in most cases the problem will be that the build will fail, because some links will be missing in old docs etc. 

So I think we are well covered.

Reply:
Makes sense, thanks for the details